video/mp_image.h

```diff
@@ -43,6 +43,7 @@
 // usually copy the whole struct, so that fields added later will be preserved.
 struct mp_image_params {
     enum mp_imgfmt imgfmt;      // pixel format
+    const char *imgfmt_name;    // pixel format name
```

Comment:
> why put this here when it can be looked up from imgfmt?

Unfortunately it is not possible. We get `pl_fmt` from libplacebo, which is in no sane way mappable directly to imgfmt or even pixfmt. There are some fourcc's that could be used for mapping, but the list isn't exhaustive.

I mention that in first comment, I don't like the amount of glue this needs to talk to libplacebo. And frankly the best way forward would be to unify the structures between both projects to avoid constant and needles mapping of them.

This value is used to pass the name to the property, shouldn't be used in any other scenario.

>  also raises the question whether the caller guarantees that it points to permanently valid memory

This is on caller to ensure.

Reply:
so `imgfmt` is unset and `imgfmt_name` contains useful data, within the same structure?
that is confusing

Reply:
Alternative is duplicate everything, would be even more confusing.

Reply:
> This value is used to pass the name to the property, shouldn't be used in any other scenario.

this should be made clear then

Reply:
made it clear in commit msg